### ISSUE TITLE
[Machine Learning] Time series modeling and forecasting using Stacked LSTM architecture

### ISSUE THREAD

**Issue by charlesndirutu33** [section-engineering-education/engineering-education#7009]
#### Proposed title of article
 Time series modeling and forecasting using Stacked LSTM architecture

#### Proposed article introduction
A time series is a collection of data points organized in successive order over time. Time series analysis involves extracting meaningful patterns and other attributes of the historical data. Time series forecasting builds a model that predicts future values based on historical data. We will build a stock price prediction model.

Stock market prediction is used to determine the future value of company stock or other financial instruments traded on an exchange. The successful prediction of a stock's future price could yield significant profit. Time series analysis comprises methods for analyzing time-series data in order to extract meaningful statistics and other characteristics of the data.

In investing companies, a time series tracks the movement of the chosen data points, such as a security’s price, over a specified period of time with data points recorded at regular intervals. Time series forecasting fits a model on a historical dataset, that is, the training set, and then using it to predict a future occurrence, which is the test set.

We will use  Stacked LSTM architecture to build a Sequential Model with multiple input and hidden layers. The trained model will be used to predict Apple's future stock prices. A Stacked LSTM architecture can be defined as an LSTM model comprised of multiple LSTM layers. An LSTM layer provides a sequence output rather than a single value output.

#### Key takeaways 
1. Initializing the TensorFlow Sequential model.
2. Time series analysis and components of Apple stock data.
3. Differencing data to be and testing using unit root testing
4. Adding stacked LSTM layers and Dense Layers.
6. Time-series Forecasting/making predictions.

#### Article quality
This article will be unique because it will have a logical step-by-step explanation. We will build the time series model using stacked LSTM layers which are more efficient compared to other models. LSTM models are able to store information over a period of time, this makes them more efficient when training a time series model which is memory intensive. The model will be accurate and be able to forecast future stock prices. We will explain how to implement stacked LSTM in detail so that a reader can easily follow. It will also explain all the components in the stock price prediction model and how to fine-tune the features to produce accurate results.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

**Comment by hectorkambow** [comment]
Good afternoon and thank you for submitting your topic to the EngEd program. 

After some careful consideration it struck us that this topic may be a bit over saturated throughout other blog sites and official documentations. We're more interested in original, practitioner-focused content that takes a deeper dive into programming-centric concepts. Please feel free to suggest an alternate topic to explore. 🚀 


We typically refrain from publishing content that is covered widely on the net or other blogs. We're more interested in original, practitioner-focused content that takes a deeper dive into programming-centric concepts. 

But in order to approve the topic it has to serve value to the larger developer community at large. Please feel free to suggest an alternate topic to explore. 🚀 


https://www.google.com/search?q=Time+series+modeling+and+forecasting+using+Stacked+LSTM+architecture&rlz=1C1CHBF_enUS891US891&sourceid=chrome&ie=UTF-8
